app/controllers/products_controller.rb

```diff
@@ -9,7 +9,7 @@ class ProductsController < ApplicationController
 
   def add_to_cart
```

Comment:
let's move all cart methods to cart_contoller, and make it in one method `update` which will receive new cart object

Reply:
I move all methods, but I have no idea how can I make it in one method

Reply:
adding to cart and removing from the cart is the same as updating the cart, you can setup cart object on the FE, and when you add/delete/edit items you just send this object to the backend and replace old cart with the new one. We can discuss it offline, it will be faster